### ISSUE TITLE
glyphs property undefined

### ISSUE THREAD

**Issue by tobinbradley** [mapbox/mapbox-gl-style-spec#441]
With 8.5 I'm getting an `Uncaught TypeError: Cannot read property 'glyphs' of undefined` when used as a dependency with mapbox-gl 0.16.0. I'm using a local font path if that makes a difference (doesn't seem to when I've fiddled with it).

``` javascript
"glyphs": "./style/bright/font/{fontstack}/{range}.pbf",
```

Running `gl-style-validate` on the style file yields no errors; there only seems to be a problem when 8.5 is used with mapbox-gl. 

The error is coming from lib/validate/validate_layout_property.js at line 20.

``` javascript
} else if (propertyKey === 'text-field' && !style.glyphs) {
```

If I comment out that entire if/then section, everything works, and if I overwrite the node_modules/mapbox-gl-style-spec folder with 8.4.2 everything works. 

Thanks for any help and all the awesome work you folks do!

**Comment by tmcw** [comment]
Can you provide your style in its entirety? The position of the glyphs property in the object could be the issue.

**Comment by jfirebaugh** [comment]
Maybe related to https://github.com/mapbox/mapbox-gl-js/issues/2407?

cc @lucaswoj

**Comment by tobinbradley** [comment]
The style file is [here](https://github.com/tobinbradley/Mecklenburg-County-GeoPortal/blob/master/app/style/bright/style.json). The glyphs property is in the object root near the top:

``` javascript
{
    "version": 8,
    "name": "Bright",
    "sources": {
        "mapbox": {
            "tiles": [
                "http://maps.co.mecklenburg.nc.us/tiles/united_states_of_america/{z}/{x}/{y}.pbf",
                "http://tiles.mcmap.org/united_states_of_america/{z}/{x}/{y}.pbf"
            ],
            "type": "vector",
            "maxzoom": 14
        },
        "parcels": {
          "type": "vector",
          "tiles": [
            "http://maps.co.mecklenburg.nc.us/tiles/parcels_and_buildings/{z}/{x}/{y}.pbf",
            "http://tiles.mcmap.org/parcels_and_buildings/{z}/{x}/{y}.pbf"
          ],
          "maxzoom": 14,
          "minzoom": 14
        }
    },
    "sprite": "./style/bright/sprite",
    "glyphs": "./style/bright/font/{fontstack}/{range}.pbf",
    "metadata": {...},
    "layers": [...]
```

If simulating the problem proves to be a problem, and you're confident the solution isn't that I need to be hit with a shovel, you can simulate it with:

```
git clone https://github.com/tobinbradley/Mecklenburg-County-GeoPortal.git
npm install -g gulp
cd Mecklenburg-County-GeoPortal
npm install
gulp build
gulp
```

Error in the javascript console and a map with a whole lotta nothing.

**Comment by tobinbradley** [comment]
A little more info. I stuck a console.log(style.sprite) in validate_layout_property.js. It came up with the right answer 161 times (?) on page load and then hit a `Uncaught TypeError: Cannot read property 'sprite' of undefined`.

**Comment by lucaswoj** [comment]
This is a bug in gl-js (https://github.com/mapbox/mapbox-gl-js/issues/2407) which will be fixed in v0.17.
